index.py

```diff
@@ -28,47 +29,18 @@ def index():
     message = ""
     return render_template('index.html', form=form, message=message)
 
+
 @app.route("/", methods=['POST'])
 def index_post():
-    form = NameForm()
-    message = ""
-    if form.validate_on_submit():
-        game_tag = form.gametag.data
-        tag_line = form.tagline.data
-        try:
-            puuid = cpmdetail.get_puuid(game_tag, tag_line)
-        except NotFoundError as ex:
-            return render_template('index.html', form=form, message=ex.message)
-        print(puuid)
-        if len(puuid) <= 0:
-            message = str(f"No account found for: {game_tag}#{tag_line}. Check for typos and try again")
-            print(message)
-            return render_template('index.html', form=form, message=message)
-
-        matches = cpmdetail.get_matches(puuid)
-        matches_length = len(matches)
-        if matches_length > 0:
-            message = str(f"Matches for: {game_tag}#{tag_line}")
-            print(message)
-            match_detail_dict = cpmdetail.get_match_detail(matches, puuid)
-
-            return render_template('matches.html', form=form, message=message, matches_length=matches_length,
-                                   match_detail_dict=match_detail_dict, matches=matches, puuid=puuid)
-        else:
-            message = str(f"{game_tag}#{tag_line} found but no recent matches. Check for typos and try again")
-            print(message)
-            return render_template('index.html', form=form, message=message)
+    return cpmdetail.get_index_post()
 
 
-@app.route("/<puuid>/match/<id>")
-def match(puuid, id):
-    message = str(f"CSM for {id}")
-    match_id = id
-    participant_id = cpmdetail.get_participant(match_id, puuid)
-    frames = cpmdetail.get_match_timeline(match_id)
-    csm_detail = cpmdetail.get_cs_per_frame(participant_id, frames)
+@app.route("/<puuid>/<participant_id>/match/<match_id>")
+def match(puuid, participant_id, match_id):
+    message = str(f"CSM for {match_id}")
+    csm_detail = cpmdetail.get_cs_per_frame(participant_id, match_id)
     print(str(f"Getting {match_id} match for {puuid}:{participant_id}"))
-    return render_template('match-detail.html', message=message, csm_detail=csm_detail, frame_length=len(frames))
+    return render_template('match-detail.html', message=message, csm_detail=csm_detail)
```

Comment:
Talking again about what is cpmdetail's job vs index.py's - I think this is handled better. "cpmdetail, gimme match info for this match." Done